beluga-ssh-service/src/lib.rs

```diff
@@ -1,58 +1,63 @@
+}
+
+impl SshService {
+    pub const fn source() -> Self {
+        SshService { port: 8022 }
+    }
+
+    pub const fn destination() -> Self {
+        SshService { port: 22 }
+    }
+
+    pub const fn port(&self) -> u16 {
+        self.port
+    }
+}
 
 impl Service for SshService {
     async fn connect(
-        &mut self,
+        &self,
         websocket_in: Sender<bytes::Bytes>,
-        mut websocket_out: Receiver<bytes::Bytes>,
+        websocket_out: Receiver<bytes::Bytes>,
         close_service: Sender<()>,
     ) -> Result<()> {
-        let mut stream =
-            TcpStream::connect::<SocketAddrV4>(SocketAddrV4::new([127, 0, 0, 1].into(), 22))
-                .await?;
+        let stream = TcpStream::connect(SocketAddrV4::new(Ipv4Addr::LOCALHOST, self.port)).await?;
 
-        let handle = tokio::spawn(async move {
-            let (mut reader, mut writer) = stream.split();
-            let mut buff = bytes::BytesMut::with_capacity(2048);
-
-            loop {
-                tokio::select! {
-                    res = reader.read_buf(&mut buff) => {
-                        res.map_err(|err| {
-                            Error::Service(std::io::Error::other(format!(
-                                "couldn't read from ssh reader cause:\"{err}\"")))
-                        })?;
-
-                        websocket_in
-                            .send(buff.clone().freeze())
-                            .await
-                            .map_err(|err| {
-                                Error::Service(std::io::Error::other(format!(
-                                    "couldn't write to websocket channel cause: \"{err}\"")))
-                            })?;
-                        buff.clear();
-                    }
-                    bytes = websocket_out.recv() => {
-                        let Some(mut data) = bytes else {
-                            return Ok::<(), Error>(());
-                        };
-                        writer.write_all_buf(&mut data).await.map_err(|err| {
-                            Error::Service(std::io::Error::other(format!(
-                                "couldn't write to ssh socket cause:\"{err}\"")))
-                        })?;
-                    }
-                }
-            }
+        let handle = tokio::spawn(serve(stream, websocket_in, websocket_out));
+
+        tokio::spawn(async move {
+            let _ = handle.await;
+            let _ = close_service.send(()).await;
         });
 
+        Ok(())
+    }
+
+    async fn bind(
+        &self,
+        websocket_in: Sender<bytes::Bytes>,
+        websocket_out: Receiver<bytes::Bytes>,
+        close_service: Sender<()>,
+    ) -> Result<()> {
+        let listener =
+            TcpListener::bind::<SocketAddrV4>(SocketAddrV4::new(Ipv4Addr::LOCALHOST, self.port))
+                .await?;
+        let (stream, addr) = listener.accept().await?;
+        tracing::info!("Got new client : {addr:?}");
```

Comment:
Structured logging instead of formatting:
```suggestion
        tracing::info!(?addr, "Got new client");
```